File: foodly.presentation/src/utils/contracts.ts
function requireCond(cond: boolean, msg: string): void {
  if (!cond) {
    throw new Error(`[PRECONDITION FAILED] ${msg}`);
  }
}

function ensure(cond: boolean, msg: string): void {
  if (!cond) {
    throw new Error(`[POSTCONDITION FAILED] ${msg}`);
  }
}

function invariant(cond: boolean, msg: string): void {
  if (!cond) {
    throw new Error(`[INVARIANT FAILED] ${msg}`);
  }
}

export { requireCond, ensure, invariant };

File: foodly.preferences/contracts/contracts.go
package contracts
import "fmt"

func Require(cond bool, msg string) {
	if !cond {
		panic(fmt.Sprintf("[PRECONDITION FAILED] %s", msg))
	}
}

func Ensure(cond bool, msg string) {
	if !cond {
		panic(fmt.Sprintf("[POSTCONDITION FAILED] %s", msg))
	}
}

func Invariant(cond bool, msg string) {
	if !cond {
		panic(fmt.Sprintf("[INVARIANT FAILED] %s", msg))
	}
}